### ISSUE TITLE
[JENKINS-35453] Allow disabling the default commit status notifier

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/github-branch-source-plugin#970]
<p>In <a href="https://issues.jenkins.io/browse/JENKINS-33151" title="GitHub commit status design is bad" class="issue-link" data-issue-key="JENKINS-33151"><del>JENKINS-33151</del></a>, giving users the option to disable the default commit status notifier was raised (in lieu of more complex configuration, such as the ability to change the context string, which is hard-coded to 'Jenkins': <a href="https://github.com/jenkinsci/github-branch-source-plugin/blob/master@%7B2016-06-08%7D/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java#L78" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/github-branch-source-plugin/blob/master@%7B2016-06-08%7D/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java#L78</a>)</p>

<p>I think there's probably a use case for this, and it would allow users with more complicated status logic (such as in that ticket) to implement their own, and just opt out of the default one.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dominics">dominics</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35453" target="_blank">Allow disabling the default commit status notifier</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/vlatombe.png" title="vlatombe's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vlatombe">vlatombe</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: github-branch-source-plugin
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2017-10-06T01:06:23+00:00
<li><b>votes</b>: 13
<li><b>watchers</b>: 27
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>In <a href="https://issues.jenkins.io/browse/JENKINS-33151" title="GitHub commit status design is bad" class="issue-link" data-issue-key="JENKINS-33151"><del>JENKINS-33151</del></a>, giving users the option to disable the default commit status notifier was raised (in lieu of more complex configuration, such as the ability to change the context string, which is hard-coded to 'Jenkins': <a href="https://github.com/jenkinsci/github-branch-source-plugin/blob/master@%7B2016-06-08%7D/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java#L78" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/github-branch-source-plugin/blob/master@%7B2016-06-08%7D/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java#L78</a>)</p>

<p>I think there's probably a use case for this, and it would allow users with more complicated status logic (such as in that ticket) to implement their own, and just opt out of the default one.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote links associated with this issue:_


  - [PR 110](https://github.com/jenkinsci/github-branch-source-plugin/pull/110)
  - [CloudBees Internal CD-310](https://cloudbees.atlassian.net/browse/CD-310)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35453?focusedId=262140&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-262140">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Cf. comment in <a href="https://issues.jenkins.io/browse/JENKINS-26522" title="Annotated block/stage status" class="issue-link" data-issue-key="JENKINS-26522"><del>JENKINS-26522</del></a>.</p>
</pre>
</details>

</ul>
</details>

<p>Cf. comment in <a href="https://issues.jenkins.io/browse/JENKINS-26522" title="Annotated block/stage status" class="issue-link" data-issue-key="JENKINS-26522"><del>JENKINS-26522</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/aehlke.png" title="aehlke's avatar" /> aehlke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35453?focusedId=275549&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-275549">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Just to chime in with one use case: we're incrementally rolling out Jenkinsfile for CI, and so we'd prefer to disable the commit status update while we're still prototyping. We want to be able to have Jenkinsfile run in parallel with our existing CI to build all branches, to test it at scale, but since it's still unproven for us we wouldn't want any issues to result in GitHub pull requests being marked as failed as we roll it out.</p>

<p>We have workarounds, just wanted to provide one use case since you seemed uncertain about the value of this.</p>
</pre>
</details>

</ul>
</details>

<p>Just to chime in with one use case: we're incrementally rolling out Jenkinsfile for CI, and so we'd prefer to disable the commit status update while we're still prototyping. We want to be able to have Jenkinsfile run in parallel with our existing CI to build all branches, to test it at scale, but since it's still unproven for us we wouldn't want any issues to result in GitHub pull requests being marked as failed as we roll it out.</p>

<p>We have workarounds, just wanted to provide one use case since you seemed uncertain about the value of this.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>miker64:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35453?focusedId=281812&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-281812">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>In my case, I want to send a multiple github statuses, with context equal to the pipeline stage, so we can use protected branches, but only lock out failing lint and unit testing stages, not our experimental stages or deploy stages.</p>

<p>I <b>believe</b> this would be a helpful for that sort of workflow.</p>
</pre>
</details>

</ul>
</details>

<p>In my case, I want to send a multiple github statuses, with context equal to the pipeline stage, so we can use protected branches, but only lock out failing lint and unit testing stages, not our experimental stages or deploy stages.</p>

<p>I <b>believe</b> this would be a helpful for that sort of workflow.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>dbaikov:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35453?focusedId=283434&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-283434">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>+1 This feature would be very useful for us!</p>

<p>We also have a need to send customized notifications to github with different context depending on the current build stage.</p>
</pre>
</details>

</ul>
</details>

<p>+1 This feature would be very useful for us!</p>

<p>We also have a need to send customized notifications to github with different context depending on the current build stage.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>myoung34:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35453?focusedId=288201&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-288201">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This bug is annoying my coworkers. We use a Jenkinsfile to test branches or deploy based on boolean parameter </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">IS_DEPLOY</pre>
</div></div>
<p> Everything works great, except if you deploy jenkins sends a pass/fail to the branch/PR which isn't what I want, i only care that the tests send that pass/fail.</p>
</pre>
</details>

</ul>
</details>

<p>This bug is annoying my coworkers. We use a Jenkinsfile to test branches or deploy based on boolean parameter </p>

<pre>
IS_DEPLOY</pre>
<p> Everything works great, except if you deploy jenkins sends a pass/fail to the branch/PR which isn't what I want, i only care that the tests send that pass/fail.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/sparklepony.png" title="sparklepony's avatar" /> sparklepony:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35453?focusedId=293619&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-293619">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'd also like to give a +1 that it's a bummer I can't turn these off. I'm using my own custom notification, so having these in there as well creates confusing clutter that I have to explain to my coworkers.</p>
</pre>
</details>

</ul>
</details>

<p>I'd also like to give a +1 that it's a bummer I can't turn these off. I'm using my own custom notification, so having these in there as well creates confusing clutter that I have to explain to my coworkers.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/allan_burdajewicz.png" title="allan_burdajewicz's avatar" /> allan_burdajewicz:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35453?focusedId=310293&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-310293">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I can see the PR has been closed. Are there still plans to implement this improvement ? <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly" class="user-hover" rel="stephenconnolly">stephenconnolly</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vlatombe" class="user-hover" rel="vlatombe">vlatombe</a></p>
</pre>
</details>

</ul>
</details>

<p>I can see the PR has been closed. Are there still plans to implement this improvement ? <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly" class="user-hover" rel="stephenconnolly">stephenconnolly</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vlatombe" class="user-hover" rel="vlatombe">vlatombe</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35453?focusedId=316127&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-316127">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>There is a plugin that can provide this as a trait to the Github branch source.</p>

<p><a href="https://github.com/bluesliverx/disable-github-multibranch-status-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/bluesliverx/disable-github-multibranch-status-plugin</a> </p>
</pre>
</details>

</ul>
</details>

<p>There is a plugin that can provide this as a trait to the Github branch source.</p>

<p><a href="https://github.com/bluesliverx/disable-github-multibranch-status-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/bluesliverx/disable-github-multibranch-status-plugin</a> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/akom.png" title="akom's avatar" /> akom:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35453?focusedId=321943&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-321943">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I didn't notice this ticket until now - I just implemented a somewhat more flexible version of this as a PR to github-branch-source in <a href="https://issues.jenkins.io/browse/JENKINS-48408" title="GitHub status updating and exposing URL should be an option" class="issue-link" data-issue-key="JENKINS-48408">JENKINS-48408</a> (oops).</p>

<p><a href="https://github.com/jenkinsci/github-branch-source-plugin/pull/169" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/github-branch-source-plugin/pull/169</a></p>

<p>The advantage is that you get an option to hide the Jenkins URL but still have notifications in github.</p>
</pre>
</details>

</ul>
</details>

<p>I didn't notice this ticket until now - I just implemented a somewhat more flexible version of this as a PR to github-branch-source in <a href="https://issues.jenkins.io/browse/JENKINS-48408" title="GitHub status updating and exposing URL should be an option" class="issue-link" data-issue-key="JENKINS-48408">JENKINS-48408</a> (oops).</p>

<p><a href="https://github.com/jenkinsci/github-branch-source-plugin/pull/169" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/github-branch-source-plugin/pull/169</a></p>

<p>The advantage is that you get an option to hide the Jenkins URL but still have notifications in github.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/carpnick2.svg" title="carpnick2's avatar" /> carpnick2:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35453?focusedId=363651&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-363651">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is unfortunate to see this isn't implemented.  I think this is a use case that is strong enough to be documented.  There is no mention on the documentation on how to turn this automatic setting off: <a href="https://go.cloudbees.com/docs/plugins/github-branch-source/" class="external-link" target="_blank" rel="nofollow noopener">https://go.cloudbees.com/docs/plugins/github-branch-source/</a></p>

<p> </p>

<p>Also most of us working with setting our own status's are using this plugin documentation: <a href="https://plugins.jenkins.io/github" class="external-link" target="_blank" rel="nofollow noopener">https://plugins.jenkins.io/github</a></p>

<p> </p>

<p>So basically we have 2 plugins at play now to do proper customized commit status's.  Now for us to turn off the default Status, we need to use a third one from <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a>.</p>

<p> </p>

<p> </p>

<p>While I realize this is resolved, one must ask, why is this level of customization (3) plugins required to send only the notifications I want to Github when using the github-branch-source plugin....?  As usual, yes make the defaults easy for new users, but asking for an ability to turn off a default (default status notification) seems prudent from the end user community trying to customize their business's workflow.</p>
</pre>
</details>

</ul>
</details>

<p>This is unfortunate to see this isn't implemented.  I think this is a use case that is strong enough to be documented.  There is no mention on the documentation on how to turn this automatic setting off: <a href="https://go.cloudbees.com/docs/plugins/github-branch-source/" class="external-link" target="_blank" rel="nofollow noopener">https://go.cloudbees.com/docs/plugins/github-branch-source/</a></p>

<p> </p>

<p>Also most of us working with setting our own status's are using this plugin documentation: <a href="https://plugins.jenkins.io/github" class="external-link" target="_blank" rel="nofollow noopener">https://plugins.jenkins.io/github</a></p>

<p> </p>

<p>So basically we have 2 plugins at play now to do proper customized commit status's.  Now for us to turn off the default Status, we need to use a third one from <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a>.</p>

<p> </p>

<p> </p>

<p>While I realize this is resolved, one must ask, why is this level of customization (3) plugins required to send only the notifications I want to Github when using the github-branch-source plugin....?  As usual, yes make the defaults easy for new users, but asking for an ability to turn off a default (default status notification) seems prudent from the end user community trying to customize their business's workflow.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/github-branch-source-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-36574%22">JENKINS-36574</a>]</i>
